geopandas/tools/overlay.py

```diff
@@ -55,8 +57,7 @@ def _extract_rings(df):
 
     return rings
 
-
-def overlay(df1, df2, how, use_sindex=True):
+def overlay_slow(df1, df2, how, use_sindex=True, **kwargs):
```

Comment:
I think it is fine to keep this initially (to eg easier compare both in the beginning if people open issues about the changed overlay), but I would make it private (eg `_overlay_old`).